src/util/records.py

```diff
@@ -1098,10 +1098,10 @@ def add_ref(ref):
 
 def delete_unused(cr, *xmlids, **kwargs):
     """
-    Remove unused records.
+    Remove unrestricted records.
 
-    This function will remove records pointed by `xmlids` only if they are not referenced
-    from any table.
+    This function will remove records pointed by `xmlids` only if they are NOT referenced
+    with `restrict` ondelete clause from any table.
 
```

Comment:
The way I see the definition of unused here, is that records are not referenced from any table or only referenced with `on delete cascade`, or have a reference from the same model that is not `on delete restrict`. 

So the set null is the same as restrict, only cascade is ignored when checking fk on other tables.

Reply:
https://github.com/odoo/upgrade-util/pull/98#issuecomment-2176405285